services/provider/server/server.go

```diff
@@ -341,77 +345,314 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
+				"csi.storage.k8s.io/node-stage-secret-namespace":        consumerResource.Status.Client.OperatorNamespace,
+				"csi.storage.k8s.io/controller-expand-secret-namespace": consumerResource.Status.Client.OperatorNamespace,
 			}
 		}
+		//TODO: Day-2 storageClasses
+
+		extR = append(extR,
+			&pb.ExternalResource{
+				Name: storageClassName,
+				Kind: "StorageClass",
+				Data: mustMarshal(&storagev1.StorageClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        storageClassName,
+						Annotations: annotations,
+						Labels:      labels,
+					},
+					ReclaimPolicy:        ptr.To(corev1.PersistentVolumeReclaimDelete),
+					AllowVolumeExpansion: ptr.To(true),
+					Provisioner:          provisioner,
+					Parameters:           parameters,
+				}),
+			},
+		)
 	}
 
-	if port < 0 {
-		return nil, fmt.Errorf("mgr pod port is empty")
+	for i := 0; i < len(consumerResource.Spec.VolumeSnapshotClasses); i++ {
+		snapshotClassName := consumerResource.Spec.VolumeSnapshotClasses[i].Name
+
+		provisioner := ""
+		labels := map[string]string{}
+		parameters := map[string]string{}
+
+		if snapshotClassName == storagecluster.GenerateNameForSnapshotClass(storageCluster, storagecluster.RbdSnapshotter) {
+			provisioner = util.RbdDriverName
+			labels = map[string]string{
+				ramenDRStorageIDLabelKey: rbdStorageID,
+			}
+			parameters = map[string]string{
+				"clusterID": rbdClientProfileName,
+				"csi.storage.k8s.io/snapshotter-secret-name":      rbdProvisionerSecretName,
+				"csi.storage.k8s.io/snapshotter-secret-namespace": consumerResource.Status.Client.OperatorNamespace,
+			}
+		} else if snapshotClassName == storagecluster.GenerateNameForSnapshotClass(storageCluster, storagecluster.CephfsSnapshotter) {
+			provisioner = util.CephFSDriverName
+			labels = map[string]string{
+				ramenDRStorageIDLabelKey: cephFsStorageID,
+			}
+			parameters = map[string]string{
+				"clusterID": cephfsClientProfileName,
+				"csi.storage.k8s.io/snapshotter-secret-name":      cephFsProvisionerSecretName,
+				"csi.storage.k8s.io/snapshotter-secret-namespace": consumerResource.Status.Client.OperatorNamespace,
+			}
+		}
+		//TODO: Day-2 snapshotclass and nfsSnapshot
+
+		extR = append(extR,
+			&pb.ExternalResource{
+				Name:   snapshotClassName,
+				Kind:   "VolumeSnapshotClass",
+				Labels: labels,
+				Data: mustMarshal(&snapapi.VolumeSnapshotClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: snapshotClassName,
+					},
+					Driver:         provisioner,
+					Parameters:     parameters,
+					DeletionPolicy: snapapi.VolumeSnapshotContentDelete,
+				}),
+			},
+		)
 	}
 
-	extR = append(extR, &pb.ExternalResource{
-		Name: "monitoring-endpoint",
-		Kind: "CephCluster",
-		Data: mustMarshal(map[string]string{
-			"MonitoringEndpoint": mgrPod.Status.HostIP,
-			"MonitoringPort":     strconv.Itoa(int(port)),
-		})})
+	for i := 0; i < len(consumerResource.Spec.VolumeGroupSnapshotClasses); i++ {
```

Comment:
This if-else approach is unreasonable for so many built in *classes. Can we move to a template based approach where the template is chosen based on the name?